mender-convert-modify

```diff
@@ -196,18 +196,18 @@ EOF
 
   run_and_log_cmd "sudo mkdir -p work/boot/EFI/BOOT"
   run_and_log_cmd "sudo cp work/${efi_name} -P work/boot/EFI/BOOT/${efi_target_name}"
+fi
 
-  # Copy dtb directory to the boot partition for use by the bootloader.
-  if [ -d work/rootfs/boot/dtbs ]; then
-    # Look for the first directory that has dtb files. First check the base
-    # folder, then any subfolders in versioned order.
-    for candidate in work/rootfs/boot/dtbs $(find work/rootfs/boot/dtbs/ -maxdepth 1 -type d | sort -V -r); do
-      if [ $(find $candidate -maxdepth 1 -name '*.dtb' | wc -l) -gt 0 ]; then
-        run_and_log_cmd "sudo cp -r $candidate work/boot/dtb"
-        break
-      fi
-    done
-  fi
+# Copy dtb directory to the boot partition for use by the bootloader.
+if [ -d work/rootfs/boot/dtbs ]; then
```

Comment:
I saw you did something similar to this in meta-mender and was going to ask about it.  Doesn't this present a risk of them being out of date after full rootfs updates are processed?  Do we want to setup a state script to update them?

Reply:
It does. But this is how the UEFI support works in U-Boot; it loads the dtb files from the UEFI partition. How the dtb files are loaded falls under the whole `distro_bootcmd` framework, and therefore it will be hard to "menderize" it by loading them from a different partition. I guess it makes sense from their perspective, you don't want a UEFI loader to look *outside* of the UEFI partition. Quite a tricky issue though....

Using a state script is a good idea, but it comes with the same caveats as updating the bootloader: There is no rollback if it breaks. So I don't think we should enable it by default, since I reckon having an old dtb file is better than having a broken one.

Ideally we should think about some way of solving this, but I think it's ok to leave it as it is for now. It's not worse than it already was, and if you're integrating with U-Boot, the DTB files will end up in both places, so you can still make DTB files upgradeable by making sure your U-Boot commands are picking them from the rootfs partition.